NSS-Day-Cohort-69/tabloid-one-more-div
# Feature - Post Details View

### Purpose
To allow users to view the contents and reactions of a Post, and conditionally render buttons in the Post Details view based on the Users' roles and relation to the Post

### Files Changed
- Created initial Reaction DTO class for use in Posts in new `ReactionForPostDTO.cs` DTO class
- Removed nav property "Reaction" from `PostReactionForPostDTO.cs`
- Created initial Post DTO class for use when viewing a singular Post in new `PostForDetailsDTO.cs`
- Added GET endpoint that returns a Post with its Category, Tags, Reactions, and count of Comments in new `PostController.cs`
- Added fetch that gets a Post and its contents (sans actual comments) in `postManager.js`
- Created file, and wrote initial rendering and logic for Post Details view in new `PostDetails.jsx` component module
- Added routing for Post Details view in `ApplicationViews.jsx`
- Added ability to navigate to a specific Post or User Profile in `PostList.jsx`

### To Test
Fetch, setup, and run according to project README, then confirm the following
- Clicking on a Posts' title in the Post List view navigates the User to the related Post Details view
- Post Details view renders appropriate Category and Tags if applicable
- Post Details view shows accurate counts for each Reaction
- Reaction button is different color when the logged in User has used that Reaction on the current Post
- The "Subscribe" button only renders in the Post Details view when the logged in User is not the Author of the Post
- The "Edit" and "Manage Tags" button only renders in the Post Details view when the logged in User is the Author of the Post
- The "Delete" button only renders in the Post Details view when the logged in User is either the Author of the Post, or an Admin

closes #42
closes #12
